### PR TITLE
[semver:minor] Make 'build-image' command more friendly

### PR DESCRIPTION
**Pull request by aaron-trout**
### Checklist

- [x] All new jobs, commands, executors, parameters have descriptions
- [x] Examples have been added for any significant new features
- [x] README has been updated, if necessary

### Motivation, issues

- I want to use the `build-image` command directly to make sure the images build on pull requests
- Some of my images are based off other private ECR images, so still need to auth

### Description

- Add a `checkout` param to the `build-image` command so we can match the API of `build-and-push-image` (but with default `false` so we don't checkout twice when using the build and push)
- Move all the AWS CLI stuff under the `ecr-login` command so that we can use it in both build and build&push.

### CODE REVIEW

**Comment by aaron-trout** [comment]
FYI I am trying to test this at the moment but I have somehow screwed up my github login on CircleCI so I can't get the job to run 😆

Just getting a load of unauthorized errors; will comment here when I have got it to work!

**Comment by aaron-trout** [comment]
> FYI I am trying to test this at the moment but I have somehow screwed up my github login on CircleCI so I can't get the job to run 😆
> 
> Just getting a load of unauthorized errors; will comment here when I have got it to work!

Whoops, forgot to comment back! This is now working in my CI pipeline using this branch of the orb (I published it under `aaron-trout/aws-ecr@dev:just-build`.

**Comment by gmemstr** [comment]
@aaron-trout I've kicked off the integration tests, but would you mind resolvingt the conflicts? :)

**Comment by aaron-trout** [comment]
Ugh, fixed those and some more cropped up in the last day! I'm going to need to figure out what all this new stuff is now to see if I need to modify any of it so please bear with.

**Comment by aaron-trout** [comment]
Ok, think I see what is going on. Basically #105 duplicated some of the work which was done on this PR by moving the AWS CLI setup stuff under the ecr login step.

**Comment by aaron-trout** [comment]
Ok rebased and dropped the duplicate changes now, so the diff here is a bit smaller now 🎉 

@gmemstr can you run the tests again? Thanks! 

I'll also try and build the orb and test it on my own repos again but I am on vacation Monday so this will not happen until Tuesday.

File: src/commands/build-image.yml [hunk 1]
```diff
@@ -77,7 +89,12 @@ parameters:
 
 steps:
   - when:
-      condition: <<parameters.skip-when-tags-exist>>
+      condition: <<parameters.checkout>>
+      steps:
+        - checkout
+
+  - when:
+      condition: <<parameters.ecr-login>> or <<parameters.skip-when-tags-exist>>
```
Comment (aaron-trout) [inline src/commands/build-image.yml:97]: Oh, I didn't actually check if this syntax is legit, this was just a guess!

Comment (aaron-trout) [reply src/commands/build-image.yml]: Ok based on https://circleci.com/docs/2.0/configuration-reference/#logic-statements, I think it is like this:

```yaml
condition:
  or:
    - expression_a
    - expression_b
```

I have updated the spec now!

**Comment by gmemstr** [comment]
Appreciate the effort being put into this! Reviewing now :)

**Comment by aaron-trout** [comment]
@gmemstr thanks for the merge, this is now working with version 6.12.0 however I am getting a couple steps duplicated when using build-image:

Screenshot 2020-07-28 at 17 33 16

I'll raise an issue and a PR for this, I think I found the issue :-)

**Comment by aaron-trout** [comment]
@gmemstr raised https://github.com/CircleCI-Public/aws-ecr-orb/pull/110 with a fix

**Pull request merged**